packages/playground/src/components/app-root/app-root.tsx

```diff
@@ -255,26 +257,50 @@ export class AppRoot {
     }
 
     const query = {
-      type: Node.MethodName.GET_MY_FREE_BALANCE_FOR_STATE,
+      type: Node.MethodName.GET_FREE_BALANCE_STATE,
       requestId: window["uuid"](),
-      params: { multisigAddress } as Node.GetMyFreeBalanceForStateParams
+      params: { multisigAddress } as Node.GetFreeBalanceStateParams
     };
 
     const { result } = await node.call(query.type, query);
 
-    const { balance } = result as Node.GetMyFreeBalanceForStateResult;
+    const { state } = result as Node.GetFreeBalanceStateResult;
+
+    // Had to reimplement this on the frontend because the method can't be imported
+    // due to ethers not playing nice with ES Modules in this context.
+    const getAddress = (xkey: string, k: number) =>
+      ethers.utils.computeAddress(
+        ethers.utils.HDNode.fromExtendedKey(xkey).derivePath(String(k))
+          .publicKey
+      );
+
+    const balances = [
+      ethers.utils.bigNumberify(state.aliceBalance),
+      ethers.utils.bigNumberify(state.bobBalance)
+    ];
+
+    const [myBalance, counterpartyBalance] = [
+      balances[
+        [state.alice, state.bob].findIndex(
+          address => address === getAddress(nodeAddress, 0)
+        )
+      ],
+      balances[
+        [state.alice, state.bob].findIndex(
+          address => address !== getAddress(nodeAddress, 0)
```

Comment:
This `nodeAddress`  has to be the other party's public identifier.

Reply:
Is it necessary? I'm basically saying: `OK, you're *not* my address derived from my xpub, so you _have_ to be the counterparty's address`

Reply:
Err didn't catch the `!==`

Yeah I suppose this is fine.